Create a FAQ link

devinreams opened the pull request:
admittedly a total copy/paste job and not certain if good for long term maintenance with potential for changes (like if/when one or the other becomes something else) versus making send-feedback abstract to take two and then split out later when either one changes to be something other than a simple link :thinking_face:

@jimporter what do you think? should we just re-purpose send-feedback for now? or split out into a separate component like this? or some other approach? I may be over complicating this...

Fixes #345

pdehaan commented on test/list/manage/components/open-faq-test.js:
```diff
@@ -0,0 +1,25 @@
+
+describe("list > manage > components > <OpenFAQ/>", () => {
+  it("FAQ link opened", () => {
+    const windowOpen = sinon.stub(window, "open");
```
If I understand correctly (and I rarely do), this is causing Travis to fail because we're doing a `sinon.stub(window, "open")` for both the feedback button tests for the Feedback and now FAQ button: https://github.com/mozilla-lockbox/lockbox-extension/blob/ed488b5317ca87d84f618a6bf32d29c6de54eda0/test/list/manage/components/send-feedback-test.js#L18

> "Attempted to wrap open which is already wrapped"

Not sure if we need a cleanup step to unstub this for each suite (if that's a thing). ¯\\\_(ツ)\_/¯

---

https://travis-ci.org/mozilla-lockbox/lockbox-extension/jobs/312183180#L5109-L5122

```
FAILED TESTS:
  list > manage > components > <SendFeedback/>
    ✖ feedback link opened
      Firefox 57.0.0 (Linux 0.0.0)
    Attempted to wrap open which is already wrapped
    checkWrappedMethod@webpack:///node_modules/sinon/lib/sinon/util/core/wrap-method.js:39:0 <- test/index.js:66490:21
    wrapMethod@webpack:///node_modules/sinon/lib/sinon/util/core/wrap-method.js:85:0 <- test/index.js:66536:13
    stub@webpack:///node_modules/sinon/lib/sinon/stub.js:60:0 <- test/index.js:35429:44
    @webpack:///test/list/manage/components/send-feedback-test.js:18:23 <- test/index.js:241843:24
    
    --------------
    wrapMethod@webpack:///node_modules/sinon/lib/sinon/util/core/wrap-method.js:110:0 <- test/index.js:66561:26
    stub@webpack:///node_modules/sinon/lib/sinon/stub.js:60:0 <- test/index.js:35429:44
    @webpack:///test/list/manage/components/open-faq-test.js:18:23 <- test/index.js:241550:24
```

https://travis-ci.org/mozilla-lockbox/lockbox-extension/jobs/312183180#L4836-L4839

```
  list > manage > components > <OpenFAQ/>
    ✔ FAQ link opened
  list > manage > components > <SendFeedback/>
    ✖ feedback link opened
```

devinreams replied:
Thanks, @pdehaan. I did see the failure but chose not to fix it until I knew if we should continue down this path.

Tests are passing again based on @jimporter's new telemetry middware approach now. ✅

devinreams commented:
>I think there is likely benefit to a "ExternalLink" component of some sort

@linuxwolf Yeah definitely. But based on @jimporter's feedback I assume we can stay on this route for now and can come back to it if we need to later?

>Please rebase this onto my telemetry patch, since there are significant changes to how we open external links in that.

@jimporter OK! Rebased over #385. I also added the new `faqClick` event to `metrics.md` and everywhere else.

<img width="746" alt="screen shot 2017-12-06 at 1 13 12 pm" src="https://user-images.githubusercontent.com/49511/33683168-3f5584d2-da87-11e7-9a39-0ea504cbe803.png">

linuxwolf commented:
Rebased branch on latest #351 and using the common `<ExternalLink/>` widget.

![open-faq](https://user-images.githubusercontent.com/757401/33789717-f12d9418-dc37-11e7-8aa9-7157fc900092.gif)

devinreams commented:
@jimporter I rebased this once again against #351 and everything is looking good ✅ 

Can you please take another look before we merge?

Heads up, @m8ttyB. This one is almost live. 🚢

jimporter commented:
@devinreams @linuxwolf The only part that's red is the coverage; if you can verify that we have coverage for everything but the telemetry bit, then this is good to go. I'd check but it's late and I'm starting to lose coherence... (It's probably fine either way, to be honest.)

linuxwolf commented:
@jimporter @devinreams 
Running coverage locally, `<ExternalLink/>` and `<OpenFAQ/>` both have 100% coverage.  It looks like the drop is still `manage` telemetry.

:shipit:

devinreams commented:
Thanks for reviewing and double checking, @jimporter @linuxwolf. 🙇
